SWIPs/swip-draft_app_data_protocols.md

```diff
@@ -0,0 +1,119 @@
+status: Draft
+type: Standards Track (Core, Networking, Interface)
+category: Interface
+created: 2019-08-06
+---
+
+<!--You can leave these HTML comments in your merged SWIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new SWIPs. Note that a SWIP number will be assigned by an editor. When opening a pull request to submit your SWIP, please use an abbreviated title in the filename, `SWIP-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SWIP.-->
+
+App developers shouldn't have to reinvent the wheel when building on top of Swarm.
+
+## Abstract
+
+<!--A short (~200 word) description of the technical issue being addressed.-->
+
+Apps built on top of Swarm share similar end-user needs, such as establishing contact with other users, sending messages, discovering files...
+
+This SWIP aims to define core data structures and protocols that can be implemented by any app using Swarm to support these needs.
+
+## Motivation
+
+<!--The motivation is critical for SWIPs that want to change the Swarm protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SWIP solves. SWIP submissions without sufficient motivation may be rejected outright.-->
+
+User data is currently exploited by technology companies with little respect for privacy and security, let alone transparency about the usage they make of this data.
+Even though data is generated by the users, it is usually not owned by them but rather the service providers that have incentives to use this as a way to lock-in their users.
+
+Swarm can provide an opportunity to shift this relationship between data ownership and services using this data by allowing users to store their data directly in Swarm, and grant access to the apps and services they choose.
+The problem is then to define data formats that can be shared by different apps and services having different purposes, and protocols to ensure compatiblity and security between all the interested parties interacting with this data.
+
+Beyond end-user incentives of data ownership, shared formats and protocols can be beneficial to application and services developers as a way to get started faster with development on top of Swarm, and possibly access to existing user data.
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for the current Swarm platform and future client implementations.-->
+
+### Scope (WIP)
+
+- Should only rely on existing features of Swarm, but might evolve as new possibilities are added
+- Should be usable via a HTTP gateway
```

Comment:
What does this mean exactly? No PSS like functionality can be used?

Reply:
I think the core protocols should work with the "minimum service" provided by Swarm, but additional functionalities could use additional Swarm capabilities as they are available.
For example we could have a communication channel between 2 contacts that uses feeds by default, but users could opt-in to send messages over PSS as well for faster interactions.
What do you think?

Reply:
Ah ok, I understand what you mean and this absolutely makes sense.